### ISSUE TITLE
Release TMail 0.3.2

### ISSUE THREAD

**Issue by chibenwa** [linagora/tmail-backend#189]
0.3.1 is affected by the following flows:

 - RabbitMQ is not reliable past the upgrade to RabbitMQ 3.8.18
 - Combined user repository is not correctly configured thus is buggy
 - Some additional performance enhancements are furthermore available

TODO

- [x] Tag a new release 0.3.2
- [x] Changelog
- [x] Update software-linagora-com
- [x] Deploy on Dev
- [x] Ask QA to test
- [x] Deploy on preprod
- [x] Run perf tests on preprod
- [x] Deploy on prod
- [x] Deploy on LNG

**Comment by Arsnael** [comment]
One should not forget when doing this upgrade to apply before deploy:

* https://github.com/apache/james-project/blob/master/upgrade-instructions.md#adding-threadid-column-to-message-metadata-tables
* https://github.com/apache/james-project/blob/master/upgrade-instructions.md#rework-message-denormalization

**Comment by chibenwa** [comment]
We also miss the upgrade instruction from Quan to add the `theadId to the ElasticSearch mapping`

**Comment by Arsnael** [comment]
Oh good catch...

**Comment by chibenwa** [comment]
Link to the changelog?

**Comment by Arsnael** [comment]
Forgot sorry... https://ci.linagora.com/linagora/lgs/software-linagora-com-website/merge_requests/27

**Comment by Arsnael** [comment]
Perf tests:

### JMAP

![screenshot--2021.07.27-14_23_40.png](https://images.zenhubusercontent.com/5ef2b3b0a7d2c63f71b1e1f7/cf6b8305-06ac-4f11-882b-5e7f8e1e257c)

### IMAP

![screenshot--2021.07.27-14_25_10.png](https://images.zenhubusercontent.com/5ef2b3b0a7d2c63f71b1e1f7/0a9fb0ac-6502-497f-bca5-58a5807ed897)

After comparing with the ones posted for release 0.3.1, it seems it's more or less equivalent for JMAP. However, it seems we are a bit slower for IMAP

**Comment by chibenwa** [comment]
Can we re-run please?

Rationals: the migration does generate a lot of duplicates on messageittable until you run compact you gonna read many sstables.

I expect a second IMAP run to perform better

**Comment by Arsnael** [comment]
we can yes

Did you run a nodetool compact on cassandra?

**Comment by chibenwa** [comment]
I did but I'm also running endurance tests for 6k users...

Hold on...

**Comment by Arsnael** [comment]
I can wait tomorrow morning, no hurry

**Comment by chibenwa** [comment]
I finished endurance tests ;-)

**Comment by Arsnael** [comment]
IMAP

![screenshot--2021.07.29-10_19_31.png](https://images.zenhubusercontent.com/5ef2b3b0a7d2c63f71b1e1f7/01e1d02b-4c7f-465f-a1d6-eaef23406211)

This is, hm, really good... I did switch back to master branch and I launched it with 2 imap pods and 0 jmap ones. I think maybe yeah the number of requests was over the roof before? But also I tend to think more that when we have 2 pods of each up, there is not enough resources on the nodes to support the limit of all pods together. So maybe our simulations are more efficient when we just test one and scale down the pods we don't use... 

Maybe I will throw some testings around to check this
